From 4ec3ea425a6cfc4c32f6a885a3aa0ebcb5b5ab27 Mon Sep 17 00:00:00 2001
From: Paul Thomson <paulthomson@google.com>
Date: Fri, 17 Apr 2026 13:39:52 +0100
Subject: [PATCH 08/12] Skip require isReadAccessAllowed etc.

Note: looks like we might be able to remove this now if we update
withMultiplatformLightClassSupport calls to the non-deprecated version.

ag/31195563
https://youtrack.jetbrains.com/issue/KT-72810
b/362329227#comment3

Upstream alternative:
https://github.com/jetbrains/kotlin/commit/ad055e95b29d8d8b46524ce8b3079b7e0792621d
---
 .../kotlin/light/classes/symbol/SymbolKotlinAsJavaSupport.kt  | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/analysis/symbol-light-classes/src/org/jetbrains/kotlin/light/classes/symbol/SymbolKotlinAsJavaSupport.kt b/analysis/symbol-light-classes/src/org/jetbrains/kotlin/light/classes/symbol/SymbolKotlinAsJavaSupport.kt
index 45ca7817b18..d0f9cc9d0d9 100644
--- a/analysis/symbol-light-classes/src/org/jetbrains/kotlin/light/classes/symbol/SymbolKotlinAsJavaSupport.kt
+++ b/analysis/symbol-light-classes/src/org/jetbrains/kotlin/light/classes/symbol/SymbolKotlinAsJavaSupport.kt
@@ -93,8 +93,8 @@ fun <T> withMultiplatformLightClassSupport(block: () -> T): T {
         return block()
     }
 
-    require(ApplicationManager.getApplication().isReadAccessAllowed) { "The method can only run inside a read action" }
-    require(!ApplicationManager.getApplication().isWriteAccessAllowed) { "The method cannot be run inside a write action" }
+    // require(ApplicationManager.getApplication().isReadAccessAllowed) { "The method can only run inside a read action" }
+    // require(!ApplicationManager.getApplication().isWriteAccessAllowed) { "The method cannot be run inside a write action" }
 
     try {
         KMP_CACHE.set(WeakHashMap())
-- 
2.54.0.rc1.513.gad8abe7a5a-goog

